File: backend/app/schemas/announcement.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class AnnouncementCreate(BaseModel):
    title: str
    content: str
    course_id: Optional[int] = None
    is_pinned: bool = False

class AnnouncementOut(BaseModel):
    id: int
    title: str
    content: str
    author_id: int
    course_id: Optional[int] = None
    is_pinned: bool = False
    created_at: Optional[datetime] = None
    author_name: Optional[str] = None
    course_title: Optional[str] = None
    model_config = {"from_attributes": True}

File: frontend-next/types/announcement.ts
// TODO: once backend exposes AnnouncementOut in OpenAPI,
// re-export from lib/api/schema.ts instead of re-declaring here.

export type Announcement = {
  id: number;
  title: string;
  body?: string | null;
  author_id?: number | null;
  author_name?: string | null;
  course_id?: number | null;
  is_pinned?: boolean | null;
  created_at?: string | null;
  updated_at?: string | null;
};
